### ISSUE TITLE
Feature request: Make MA0197 opt-in

### ISSUE THREAD

**Issue by Timovzl** [meziantou/Meziantou.Analyzer#1141]
# MA0197 - Do not use inheritdoc on types

Although I have opted out, I'd like to make the case that this is opinionated enough that it should be opt-in rather than opt-out, i.e. disabled by default, because the use case is so valid.

I will highly some use cases. I'll use simple summaries for brevity, but in practice, sometimes they span a few paragraphs plus type parameter documentation.

All in all, I believe inheritdoc on types is common, sensible, and defensible.

### Interface plus implementation

For Inversion of Control purposes, we create an interface for our service and program against the interface. We want to see the summary when hovering either type (it applies equally), and we do not want to maintain any duplicates.

```cs
/// <summary>
/// Blah blah blah.
/// </summary>
public interface ICustomerFactory {}

/// <inheritdoc/>
public class CustomerFactory : ICustomerFactory {}
```

### Stacked levels of abstraction

Often, an abstraction is taken a level further, but does not introduce enough novelty to warrant a different summary. We want to see the original summary, without maintaining any duplicates.

Cases include:
- We have both `IAbstraction` and `Abstraction`.
- We have both `Abstraction` and `Abstraction<T>`.
- We have both `Abstraction` and `SlightlyFurtherAbstraction`.
- We wrap a library's abstraction in our own (perhaps even _just in case_, to have an easy future extension point) and still desire its well-thought-out summary.

```cs
/// <summary>
/// An entity is a data model that is defined by its identity and a thread of continuity. It may be mutated during its life cycle.
/// </summary>
public abstract class Entity : DomainObject, IEntity {}

/// <inheritdoc/>
public abstract class Entity<TId> : Entity {}
```

**Comment by meziantou** [comment]
I changed it to suggestion by default and update the description. I also added new validation rules for inheritdoc ([MA0198](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0198.md) and [MA0199](https://github.com/meziantou/Meziantou.Analyzer/blob/main/docs/Rules/MA0199.md))

> Often, an abstraction is taken a level further, but does not introduce enough novelty to warrant a different summary

Maybe the issue is that developers create to many abstractions 🤔

**Comment by Timovzl** [comment]
Nice work adding MA0198 and MA0199. They catch real mistakes.

We can close if you're satisfied with the resolution.

I stand by the request to disable MA0197 by default, because what it catches is something intentional, not a mistake. To someone who cleans up after themselves, leaving suggestions unhandled is as almost as bad as leaving warnings unhandled. In conclusion, I'm still required to disable the analyzer manually, and I'm convinced that most people are best served doing the same - hence my fighting for a different default.

**Comment by meziantou** [comment]
I will close the issue for now, and reevaluate the default severity when I get more feedback.
